### PR TITLE
Create quote block

### LINKED ISSUE #27: Implement quote block
Original: https://www.esri.com/en-us/capabilities/imagery-remote-sensing/capabilities/management
New: https://main--esri--aemsites.hlx.page/en-us/capabilities/imagery-remote-sensing/capabilities/management

image

### PR DESCRIPTION
**Pull request by vladciocoiu**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #27 

Test URLs:
- Original: https://www.esri.com/en-us/capabilities/imagery-remote-sensing/capabilities/management
- Before: https://main--esri--aemsites.hlx.live/drafts/vciocoiu/quote
- After: https://add-quote-block--esri--aemsites.hlx.live/drafts/vciocoiu/quote

### CODE REVIEW

File: blocks/quote/quote.css [hunk 1]
```diff
@@ -0,0 +1,116 @@
+body {
+  background-color: #2B2B2B;
```
Comment (alexcarol) [inline blocks/quote/quote.css:2]: This looks like a general style, can you move it to styles.css?

File: blocks/quote/quote.css [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  .cards-card-body {
+    box-sizing: border-box;
+    padding: 3rem 0.5rem 1.5rem 1rem;
+    font-family: var(--calcite-sans-family);
+    width: 1440px;
+    max-width: 96vw;
+
+    .quote-icon {
+      --calcite-ui-icon-color: #009AF2;
```
Comment (tim-white-esri) [inline blocks/quote/quote.css:34]: Color should be `var(--calcite-ui-brand)`
I've added the Calcite CSS file in another pull request.

**Comment by alexcarol** [comment]
I observe some differences in the fonts in mobile, some might be font weights, but there might be also font aliasing differences pottentially.

Original:
image

New:
image

There were some minor differences in margins in desktop, so you might want to give that a quick review, but nothing major, so if you see it getting complex we can polish down the line based on customer feedback.

**Comment by vladciocoiu** [comment]
> I observe some differences in the fonts in mobile, some might be font weights, but there might be also font aliasing differences pottentially.

I know fonts look a bit different, but I made sure weights / families / sizes are the same. No idea why they are like that.

File: blocks/quote/quote.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { createOptimizedPicture, decorateIcons } from '../../scripts/aem.js';
+import { span } from '../../scripts/dom-helpers.js';
+
+export default function decorate(block) {
+  /* change to ul, li */
+  const ul = document.createElement('ul');
```
Comment (alexcarol) [inline blocks/quote/quote.js:6]: if you use the dom-helpers you can add the li elements directly in the constructor, with something like:
```
ul(...[block.children].map((row) => <row manipulation logic>)));
```

File: blocks/quote/quote.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      if (
+        divElement.children.length === 1
+        && divElement.querySelector('picture')
+      ) {
+        divElement.className = 'cards-card-image';
+      } else divElement.className = 'cards-card-body';
+    });
+    ul.append(li);
+  });
+  ul.querySelectorAll('img').forEach((img) => img
```
Comment (alexcarol) [inline blocks/quote/quote.js:20]: Why is this needed?

File: blocks/quote/quote.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  [...block.children].forEach((row) => {
+    const li = document.createElement('li');
+    while (row.firstElementChild) li.append(row.firstElementChild);
+    [...li.children].forEach((divElement) => {
+      if (
+        divElement.children.length === 1
+        && divElement.querySelector('picture')
+      ) {
+        divElement.className = 'cards-card-image';
+      } else divElement.className = 'cards-card-body';
```
Comment (alexcarol) [inline blocks/quote/quote.js:16]: why did you choose cards-card-body and cards-card-image as class names?

Comment (vladciocoiu) [reply blocks/quote/quote.js:16]: didn't think to change them after copying that part of the code

I will change to quote-block-image and quote-block-body

File: blocks/quote/quote.js (hunk 4 above)
Comment (vladciocoiu) [reply blocks/quote/quote.js:20]: I'm not sure, I just noticed it was also used in other places.

Should I remove it?

File: blocks/quote/quote.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      && divElement.querySelector('picture')
+    ) {
+      divElement.className = 'quote-block-image';
+    } else divElement.className = 'quote-block-body';
+
+    return divElement;
+  }))));
+
+  block.textContent = '';
+  block.append(list);
```
Comment (alexcarol) [inline blocks/quote/quote.js:17]: How about this?
```suggestion
  block.replaceWith(list);
```

Comment (vladciocoiu) [reply blocks/quote/quote.js:17]: I should probably also add all attributes of the block to the list then

Comment (alexcarol) [reply blocks/quote/quote.js:17]: fair enough

Comment (alexcarol) [reply blocks/quote/quote.js:17]: replaceChildren tends to work well in those cases, but this is good

**Review by alexcarol — APPROVED** [review]
